src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -163,50 +172,40 @@ public void SetDefaultHubName(string hubName)
             this.defaultHubName = hubName;
         }
 
-        internal string GetDebugString()
+        internal void TraceConfiguration(EndToEndTraceHelper traceHelper)
         {
-            var sb = new StringBuilder(4096);
-            sb.AppendLine("Initializing extension with the following settings:");
-            sb.Append(nameof(this.HubName)).Append(":").Append(this.HubName).Append(", ");
-
-            this.AppendStorageProviderValuesToDebugString(sb);
-
-            sb.Append(nameof(this.MaxConcurrentActivityFunctions)).Append(": ").Append(this.MaxConcurrentActivityFunctions).Append(", ");
-            sb.Append(nameof(this.MaxConcurrentOrchestratorFunctions)).Append(": ").Append(this.MaxConcurrentOrchestratorFunctions).Append(", ");
-            sb.Append(nameof(this.ExtendedSessionsEnabled)).Append(": ").Append(this.ExtendedSessionsEnabled).Append(", ");
-            sb.Append(nameof(this.UseGracefulShutdown)).Append(": ").Append(this.UseGracefulShutdown).Append(", ");
-            if (this.ExtendedSessionsEnabled)
-            {
-                sb.Append(nameof(this.ExtendedSessionIdleTimeoutInSeconds)).Append(": ").Append(this.ExtendedSessionIdleTimeoutInSeconds).Append(", ");
-            }
-
-            if (this.NotificationUrl != null)
+            // Format the options data as JSON in a way that is friendly for technical humans to read.
+            JObject configurationJson = JObject.FromObject(
+                this,
+                new JsonSerializer
+                {
+                    Converters = { new StringEnumConverter() },
+                    ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                });
+
+            // Don't trace the notification URL query string since it may contain secrets.
+            // This is the only property which we expect to contain secrets. Everything else should be *names*
```

Comment:
I'm concerned that this may not always be true, and we could eventually start logging secrets without realizing it. I'm not sure of a safe/more discoverable way of handling this. We could  implicitly treat everything as secrets unless it is tagged otherwise, but that has it's own pains.

Reply:
If this data is populated by host.json then it should never have secrets. It actually doesn't even make sense for `NotificationUrl` to be here. I guess I could solve this by removing it.

Reply:
Can we safely remove this? That seems like a breaking change.

Reply:
Is that safe to do? Seems like a breaking change.

Reply:
It is a breaking change, but I would argue there is no valid scenario for referencing this property. The documentation even mentions that this is for internal use.

Reply:
I went down the path of moving `NotificationUrl` from `DurableTaskOptions` to `DurableTaskExtension` in a previous commit. I was almost successful except that getting a final test passing would require some additional DI interfaces, which felt way out of scope for this PR. Perhaps I can consider it for a different release.